app/modules/users/components/users_home_layout.svelte

```diff
@@ -0,0 +1,88 @@
+<script>
+  import UsersHomeNav from '#users/components/users_home_nav.svelte'
+  import UserProfile from '#users/components/user_profile.svelte'
+  import GroupProfile from '#groups/components/group_profile.svelte'
+  import NetworkUsersNav from '#users/components/network_users_nav.svelte'
+  import PublicUsersNav from '#users/components/public_users_nav.svelte'
+  import { serializeUser } from '#users/lib/users'
+  import app from '#app/app'
+  import { setContext } from 'svelte'
+  import { writable } from 'svelte/store'
+  import { onChange } from '#lib/svelte/svelte'
+  import { getViewportHeight } from '#lib/screen'
+  import { resizeObserver } from '#lib/components/actions/resize_observer'
+
+  export let user, group, shelf, section, subsection = 'inventory'
+
+  const { loggedIn } = app.user
+
+  if (user?._id === app.user.id) section = 'user'
+
+  const focusStore = writable({})
+  setContext('focus-store', focusStore)
```

Comment:
I thought the consensus obtained after hours of undocumented conversation was to agree on context based variable but to forbid writable stores... and boum few months later, conversation is forgotten and look at what is happening.
I know it makes it easier to write, but side-effects of such `$focusStore = { type: 'shelf', doc: selectedShelf }` (passing the whole document (!)) can get out of control. This ticking bomb is hard to swallow for me.

Reply:
It's a great dev tool: you share state with all the children, and then when time comes to refactor, you can search which component sets the store value (`$focusStore =`) and which react to it (typically in `$:.*$focusStore`), and if the store and/or context does not seem necessary after all, refactor accordingly. As I now came to refactoring, it appears that the `$focusStore.doc` wasn't even used, so there is no more whole document passing. I tried refactoring without `getContext`/`setContext` in 96f6aac7f: we go from 18 occurrences of the word `focusStore` to 37, making the whole thing more explicit (good) but also more noisy/less grepable (not good). I'm fine going with the explicit version for now, but I don't think that it's such a foot gun that should be locked away: with some conventions on namespacing to guaranty grepability, that solves some real pain points in deeply nested layouts.

Reply:
Im sure you know my position: maybe we dont need deeply nested layouts in the first place, and when we do, better show it everywhere with lots of warnings and line of code than implying that this component is simple, but if you find a variable undeclared in the file the you just need to grep the whole codebase find where things are set. 
It's not a great dev tool for me, it hides complexity, and we should not do that for variables used in a lots of different components, cuz that that sound like a future lengthy debugging (not difficult, but lengthy).

PS: we had that conversation and we agreed that writable stores shall be banned from our codebase, dont remember it? It was at the time we wrote this framasoft article.

Reply:
Note that im far from being against stores (`$user` is great, although should probably be renamed `$currentUser`) but writable is a different matter. Another fear of mine is that the document passed through the `focusStore`, as it is difficult to find what make `doc` valid (aka how its formatted, what it should/must contain). I would feel like postponing writtable until some typescript is around to type this wild multi-purpose`doc`.

Reply:
Also, no matter of the outcome of this conversation, I would be for renaming `focusStore` -> `focusSection`

Reply:
renamed in 156136a